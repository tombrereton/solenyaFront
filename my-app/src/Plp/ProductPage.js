import React, {Component} from 'react';
import Products from './Products';
import fetchDataToJSON from '../DataAccess/DataFetcher';
import {ProductServiceEndpoint} from '../Config';

class ProductPage extends Component {
  constructor() {
    super();

    this.state = {
      productElements: []
    };
  }

  componentDidMount() {
    fetch(ProductServiceEndpoint)
      .then(resp => resp.json())
      .then(productElements => this.setState({productElements}))
      .catch(error => console.log("Error fetching in product page:",error));
  }

  render() {
    return (
      <Products productElements={this.state.productElements} /> 
    )
  }
}


export default ProductPage;
