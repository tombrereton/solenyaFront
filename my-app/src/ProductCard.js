import React from 'react';
import PropTypes from 'prop-types';

const ProductCard = (props) => {
	return (
  	<div  className="productCard" style={{margin: '1em'}}>
  	  <img width="150" src={props.splash_img_url} alt={props.product_name}/>
      <div>
        <div className="productName" style={{fontWeight: 'bold'}}>
        	{props.product_name}
        </div>
        <div className="productPrice">
        	{props.price}
        </div>
      </div>
  	</div>	
  );
};

ProductCard.propTypes ={
  splash_img_url: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  product_name: PropTypes.string.isRequired,
}

export default ProductCard;
