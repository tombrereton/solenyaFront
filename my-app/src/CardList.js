import React from 'react';
import ProductCard from './ProductCard';

const CardList = props => (
	<div>
		{props.cards.map((card,index) => <ProductCard {...card} key={index} />)}
	</div>
)

export default CardList;