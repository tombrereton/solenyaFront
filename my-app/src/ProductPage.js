import React from 'react';
import productData from './ProductData';
import ProductCard from './ProductCard';

const CardList = (props) => {
	return (
  	<div>
    	{props.cards.map((card,index) => <ProductCard {...card} key={index} />)}
    </div>
  )
};

const ProductPage = (props) => (
        <CardList cards={productData}/>
)

export default ProductPage;
