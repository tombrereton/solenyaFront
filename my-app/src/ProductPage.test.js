import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import ProductPage from './ProductPage';
import productData from './ProductData';
import ProductCard from './ProductCard';
import CardList from './CardList';


// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });


describe('ProductPage', () => {

  it('renders all products', () => {
    const wrapper = shallow(<CardList cards = {productData} />);
    const products = wrapper.find(ProductCard);
    expect(products.length).toEqual(productData.length);
  });
});


