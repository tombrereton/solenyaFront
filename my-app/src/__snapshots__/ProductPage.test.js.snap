// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard It should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt={undefined}
          src={undefined}
          width="150"
/>,
        <div>
          <div
                    className="productName"
                    style={
                              Object {
                                        "fontWeight": "bold",
                                      }
                    }
          />
          <div
                    className="productPrice"
          />
</div>,
      ],
      "className": "productCard",
      "style": Object {
        "margin": "1em",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "src": undefined,
          "width": "150",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="productName"
              style={
                            Object {
                                          "fontWeight": "bold",
                                        }
              }
/>,
            <div
              className="productPrice"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "productName",
              "style": Object {
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "productPrice",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt={undefined}
            src={undefined}
            width="150"
/>,
          <div>
            <div
                        className="productName"
                        style={
                                    Object {
                                                "fontWeight": "bold",
                                              }
                        }
            />
            <div
                        className="productPrice"
            />
</div>,
        ],
        "className": "productCard",
        "style": Object {
          "margin": "1em",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "src": undefined,
            "width": "150",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="productName"
                style={
                                Object {
                                                "fontWeight": "bold",
                                              }
                }
/>,
              <div
                className="productPrice"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "productName",
                "style": Object {
                  "fontWeight": "bold",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "productPrice",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
